src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DraftReferralService.kt

```diff
@@ -484,6 +508,108 @@ class DraftReferralService(
     }
   }
 
+  private fun createProbationPractitionerDetails(draftReferral: DraftReferral, referral: Referral) {
+    if (saveProbationPractitionerDetails) {
+      referral.probationPractitionerDetails = probationPractitionerDetailsRepository.save(
+        ProbationPractitionerDetails(
+          id = UUID.randomUUID(),
+          referral = referral,
+          nDeliusName = draftReferral.nDeliusPPName,
+          nDeliusEmailAddress = draftReferral.nDeliusPPEmailAddress,
+          nDeliusPDU = draftReferral.nDeliusPPPDU,
+          name = draftReferral.ppName,
+          emailAddress = draftReferral.ppEmailAddress,
+          pdu = draftReferral.ppPdu,
+          probationOffice = draftReferral.ppProbationOffice!!,
+        ),
+      )
+      referralRepository.save(referral)
+      checkAndMeasureDeliusData(draftReferral)
+    }
+  }
+
+  private fun checkAndMeasureDeliusData(draftReferral: DraftReferral) {
+    val eventName = "probationPractitionerDetailsLookUp"
+    when (draftReferral.hasValidDeliusPPDetails) {
+      true -> telemetryClient.trackEvent(
+        eventName,
+        mapOf(
+          "result" to "valid delius details present",
+          "referralId" to draftReferral.id.toString(),
+        ),
+        null,
+      )
+      false -> {
+        telemetryClient.trackEvent(
+          eventName,
+          mapOf(
+            "result" to "valid delius details not present",
+            "referralId" to draftReferral.id.toString(),
+          ),
+          null,
+        )
+        if (draftReferral.ppName == draftReferral.nDeliusPPName) {
+          telemetryClient.trackEvent(
+            eventName,
+            mapOf(
+              "result" to "delius name and user name matches",
```

Comment:
Could the if-else statements be refactored by doing the checks in the mapping? 

```suggestion
              "result" to if (draftReferral.ppName == draftReferral.nDeliusPPName) "delius name and user name matches" else "delius name and user name does not match",
```